packages/plugin-dom-editable/src/EventNormalizer.ts

```diff
@@ -1994,14 +1981,118 @@ export class EventNormalizer {
+        if (this._mousedownInEditable && ev.target instanceof Element) {
+            try {
+                this._sliceEventStack(check);
+
+                // When the users clicks in the DOM, the range is set in the next
+                // tick. The observation of the resulting range must thus be delayed
+                // to the next tick as well. Store the data we have now before it
+                // gets invalidated by the redrawing of the DOM.
+                this._initialCaretPosition = this._getEventCaretPosition(ev);
+                const pointerSelectionTimeout = new Timeout<EventBatch>(() => {
+                    const selectionBatch = this._getSelectionBatchOnce(ev);
+                    return selectionBatch;
+                });
+                this._triggerEventBatch(pointerSelectionTimeout.promise);
+                this._selectionTimeouts.push(pointerSelectionTimeout);
+            } catch (e) {
+                this._mousedownInEditable = false;
+                this._initialCaretPosition = undefined;
+            }
+        } else if (ev.target instanceof Element && !!ev.target.closest('[contentEditable=true]')) {
+            // When within a contenteditable element but in a non-editable
+            // context, prevent a collapsed selection by removing all ranges.
+            // TODO: remove them from the VDocument as well.
+            const pointerSelectionTimeout = new Timeout<EventBatch>(() => {
+                const selection = this._getSelection();
+                if (selection) {
+                    const collapsed =
+                        selection.anchorNode === selection.focusNode &&
+                        selection.anchorOffset === selection.focusOffset;
+                    const target = this._getClosestElement(selection.focusNode);
+                    if (collapsed && !!target.closest('[contentEditable=true]')) {
+                        document.getSelection().removeAllRanges();
+                    }
+                }
+                return {
+                    actions: [],
+                };
+            });
+            this._triggerEventBatch(pointerSelectionTimeout.promise);
+            this._selectionTimeouts.push(pointerSelectionTimeout);
+        }
+    }
+    /**
+     * Middleware betwen the `_triggerEventBatchOutside` to always capture the
+     * last action being done when sending a batch.
+     */
+    _triggerEventBatch(batchPromise: Promise<EventBatch>): void {
+        this._triggerEventBatchOutside(
+            batchPromise.then(batch => {
+                if (batch.actions.length) {
+                    this._lastAction = batch.actions[batch.actions.length - 1];
+                }
+                return batch;
+            }),
+        );
+    }
+
+    /**
+     * Check wether the last action is a selection and the selection is the
+     * different than the one provided.
+     */
+    private _isLastSelectionDifferent(selection: DomSelectionDescription): boolean {
```

Comment:
Du coup ce mécanisme-ci remplace `this.selectionHasChanged` alors ?

Reply:
Oui. `selectionHasChanged` n'existait que pour le `selectAll`.
Maintenant c'est plus **générique**. Lorsqu'on a du observer une sélection pour potentiellement l'envoyer, on regarde que ce ne soit pas la même que la dernière sinon on renvoie un batch vide.